drivers/timer/stm32_lptim_timer.c

```diff
@@ -128,6 +128,10 @@ int sys_clock_driver_init(const struct device *dev)
 		    lptim_irq_handler, 0, 0);
 	irq_enable(DT_IRQN(DT_NODELABEL(lptim1)));
 
+#ifdef CONFIG_SOC_SERIES_STM32WLX
+	__HAL_LPTIM_LPTIM1_EXTI_ENABLE_IT();
```

Comment:
I'm not clear on this addition. Why is that required only for WL ?
Also, if really needed, using LL API would be cleaner.

Reply:
I'm not too sure but it seems like that's how LPTIM1 wakeup works on WL series, think it has something to do with the dual core option. The EXTI channels says "LPTIMx wakeup".

![x](https://user-images.githubusercontent.com/891546/124500177-31d92480-ddb7-11eb-8f6b-85fbb27a1e30.png)

I see that HAL function defined for other architectures though, specifically G0, L4, L5 and WB. Not sure why it's not needed for those, but the WL I have does not wake up from STOP mode without that bit enabled.

Switching to the LL API as suggested, just added a comment around it for the moment since that API has opaque defines, let me know if you think should structure it a bit more, like adding a `#define` for the line similarly to https://github.com/zephyrproject-rtos/zephyr/blob/main/drivers/counter/counter_ll_stm32_rtc.c#L52.

Reply:
ok, thanks. Let's keep it as is for now.